openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -120,27 +152,77 @@ func InjectNamespaceWithSubject(resourceNamespace string, subjectNamespace strin
 	}
 }
 
-func LoadServingMonitoringPlatformManifests(ns string) ([]mf.Manifest, error) {
-	rbacPath := os.Getenv(servingSMRbacManifestPath)
+func getMonitoringPlarformNanifests(ns string, cType KComponentType) ([]mf.Manifest, error) {
+	rbacPath := os.Getenv(smRbacManifestPath)
 	if rbacPath == "" {
 		return nil, fmt.Errorf("failed to get the Serving sm rbac manifest path")
 	}
 	rbacManifest, err := mf.NewManifest(rbacPath)
 	if err != nil {
 		return nil, err
 	}
-	for c := range servingComponents {
-		smManifest, err := constructServiceMonitorResourceManifests(c, ns)
+	switch cType {
+	case Serving:
+		// Serving has one common sa for all pods
+		crbM, err := createClusterRoleBindingManifest("controller", ns)
 		if err != nil {
 			return nil, err
 		}
-		if smManifest != nil {
-			rbacManifest = rbacManifest.Append(*smManifest)
+		rbacManifest = rbacManifest.Append(*crbM)
+		for c := range servingComponents {
+			smManifest, err := constructServiceMonitorResourceManifests(c, ns)
+			if err != nil {
+				return nil, err
+			}
+			if smManifest != nil {
+				rbacManifest = rbacManifest.Append(*smManifest)
+			}
+		}
+	case Eventing:
+		// Only mt-broker-controller has a different than its name sa (eventing-controller)
+		for sa := range eventingComponents {
+			if sa == "mt-broker-controller" {
+				continue
+			}
+			crbM, err := createClusterRoleBindingManifest(sa, ns)
+			if err != nil {
+				return nil, err
+			}
+			rbacManifest = rbacManifest.Append(*crbM)
+		}
+		for c := range eventingComponents {
+			smManifest, err := constructServiceMonitorResourceManifests(c, ns)
+			if err != nil {
+				return nil, err
+			}
+			if smManifest != nil {
+				rbacManifest = rbacManifest.Append(*smManifest)
+			}
```

Comment:
same block as: https://github.com/openshift-knative/serverless-operator/pull/887/files#diff-9441c1af1419ac92923cabd71411c98c7702acbdad9baec26fddf7a8a4bb221bR173-R179

perhaps move to function?

Reply:
Yes could be done.

Reply:
let's do it ?

Reply:
done